File: ruby/question3/question3.rb
#!/usr/bin/env ruby

product = 1
numbers = []

# read numbers from stdin line by line
ARGF.each_line do |line|
  number = line.to_i
  # store in array
  numbers.push(number)
  # calculate running product
  product *= number
  # last?
  break if ARGF.eof?
end

# iterate numbers and print product without current number
numbers.each do |number|
  puts product / number
end

File: python/question3/question3.py
#!/usr/bin/env python

import fileinput

numbers = []
product = 1

# read numbers line by line from stdin and insert into array
# and calculate running product
for line in fileinput.input():
    number = int(line)
    numbers.append(number)
    product *= number

# print products without own number iterating array
for number in numbers:
    print product / number
